### PR TITLE
Csharp SDK for Microsoft.StorageSync Resource Provider for ApiVersion package-2018-04-02

### PR DESCRIPTION
**Pull request by ankushbindlish2**
Csharp SDK for Microsoft.StorageSync Resource Provider
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/readme.md

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/StorageSync/Management.StorageSync/Generated/SdkInfo_MicrosoftStorageSync.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        // BEGIN: Code Generation Metadata Section
+        public static readonly String AutoRestVersion = "latest";
+        public static readonly String AutoRestBootStrapperVersion = "autorest@2.0.4283";
+        public static readonly String AutoRestCmdExecuted = "cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=H:\\csharpsdk\\azure-sdk-for-net\\src\\SDKs";
+        public static readonly String GithubForkName = "Azure";
+        public static readonly String GithubBranchName = "master";
+        public static readonly String GithubCommidId = "7b95fd70c37943bd88eac39b3e7f1e4d4ab295ed";
+        public static readonly String CodeGenerationErrors = "";
+        public static readonly String GithubRepoName = "azure-rest-api-specs";
+        // END: Code Generation Metadata Section  }
```
Comment (dsgouda) [inline src/SDKs/StorageSync/Management.StorageSync/Generated/SdkInfo_MicrosoftStorageSync.cs]: Either regenerate the code or remove the `}` at the end of this line

File: src/SDKs/_metadata/storagesync_resource-manager.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (dsgouda) [inline src/SDKs/_metadata/storagesync_resource-manager.txt]: Please pull latest changes from remote psSdkJson6, run msbuild build.proj and regenerate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]
left a few comments

**Comment by dsgouda** [comment]
Also, please link the REST spec PR here

**Comment by ankushbindlish2** [comment]
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/readme.md

File: src/SDKs/StorageSync/Management.StorageSync/Microsoft.Azure.Management.StorageSync.csproj [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<PackageId>Microsoft.Azure.Management.StorageSync</PackageId>
+		<Description>Provides management capabilities for Azure Storage Sync.</Description>
+		<Version>2.0.0</Version>
```
[the remark this answers is not in the document]
Comment (ankushbindlish2) [reply src/SDKs/StorageSync/Management.StorageSync/Microsoft.Azure.Management.StorageSync.csproj]: resolved

File: src/SDKs/StorageSync/StorageSync.Tests/StorageSync.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>StorageSync.Tests</PackageId>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/StorageSync/StorageSync.Tests/StorageSync.Tests.csproj]: Please replace this with `<Version>1.0.0<Version>`

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Review by dsgouda — CHANGES_REQUESTED** [review]
Running generate.ps1 also generates a .txt file please check in this file too

**Comment by shahabhijeet** [comment]
@ankushbindlish2 please always include link to the spec PR. Having a PR link helps rather than the commit.

File: src/SDKs/StorageSync/Management.StorageSync/AzSdk.RP.props [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (shahabhijeet) [inline src/SDKs/StorageSync/Management.StorageSync/AzSdk.RP.props]: @ankushbindlish2 remove this file, this seems redundant

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/StorageSync/Management.StorageSync/AzSdk.RP.props (hunk 5 above)
Comment (ankushbindlish2) [reply src/SDKs/StorageSync/Management.StorageSync/AzSdk.RP.props]: I see it in other resource providers. Please let me know if it is ok to keep it.

**Review by shahabhijeet — APPROVED** [review]
once the CI passes, this is good to be merged.

**Pull request merged**